### PR TITLE
feat(cli): port functions list 

### PR DESCRIPTION
**Pull request by 7ttp**
## TL;DR 
 
ports functions list to native ts

## whats introduced? 
ports supabase functions list on the legacy cli path
replacing the go backed behavior with a fully ts implementation....

## ref:
- towards CLI-1319

### CODE REVIEW

File: apps/cli/src/legacy/commands/functions/list/list.handler.ts [hunk 1]
```diff
@@ -1,12 +1,163 @@
...
+      yield* output.raw(encodeYaml(functions.map(toGoYamlFunction)));
+      return;
+    }
+    if (goFmt === "toml") {
+      yield* output.raw(encodeToml({ functions: functions.map(toGoTomlFunction) }) + "\n");
+      return;
+    }
+
+    if (output.format === "json" || output.format === "stream-json") {
+      yield* output.success("", { functions });
```
[the remark this answers is not in the document]
Comment (7ttp) [reply apps/cli/src/legacy/commands/functions/list/list.handler.ts:121]: Fixed in 14b882a2 explicit output pretty now wins before TS json and stream json so mixed flag behavior matches Go again

File: apps/cli/src/legacy/commands/functions/list/list.handler.ts [hunk 2]
```diff
@@ -1,12 +1,428 @@
...
+  const output = yield* Output;
+  const goOutputFlag = yield* LegacyOutputFlag;
+  const api = yield* LegacyPlatformApi;
+  const resolver = yield* LegacyProjectRefResolver;
+  const linkedProjectCache = yield* LegacyLinkedProjectCache;
+  const telemetryState = yield* LegacyTelemetryState;
+  let resolvedProjectRef = Option.none<string>();
+
+  yield* Effect.gen(function* () {
+    const ref = yield* resolver.resolve(flags.projectRef).pipe(
```
[the remark this answers is not in the document]
Comment (avallete) [reply apps/cli/src/legacy/commands/functions/list/list.handler.ts:45]: resolver.resolve() gates the prompt on tty.stdinIsTty && output.interactive, whereas Go's ParseProjectRef prompts on stdin-TTY alone. Real-ish, but it lives in the shared LegacyProjectRefResolver, not this handler. If it's a genuine gap it affects every ported command, and I think that's a wanted divergence not a reason to block this port.

File: apps/cli/src/legacy/commands/functions/list/list.command.ts [hunk 3]
```diff
@@ -14,5 +17,21 @@ export type LegacyFunctionsListFlags = CliCommand.Command.Config.Infer<typeof co
...
+      description: "List all deployed functions in a specific project",
+    },
+  ]),
+  Command.withHandler((flags) =>
+    legacyFunctionsList(flags).pipe(
+      withLegacyCommandInstrumentation({ flags, safeFlags: ["project-ref"] }),
+      withJsonErrorHandling,
+    ),
+  ),
+  Command.provide(legacyManagementApiRuntimeLayer(["functions", "list"])),
```
[the remark this answers is not in the document]
Comment (avallete) [reply apps/cli/src/legacy/commands/functions/list/list.command.ts:36]: Falling back to the built-in profile instead of erroring is legacyManagementApiRuntimeLayer / legacy-profile.ts behavior, again shared and identical across siblings, I think this is also a wanted divergence.

File: apps/cli/src/legacy/commands/functions/list/list.handler.ts
Comment (avallete) [inline apps/cli/src/legacy/commands/functions/list/list.handler.ts:1]: **nitpick**

Extract the pure logic and unit-test it.

`list.handler.ts` is ~430 lines. It inlines a lot that the guide says should live outside the handler: 3 tagged error classes, a full hand-rolled JSON validator/parser, a unix-milli timestamp formatter, and three Go-shaped encoders (`toGoJsonFunction` / `toGoYamlFunction` / `toGoTomlFunction`).

The `apps/cli` guide prescribes the `.errors.ts` / `.format.ts` / `.encoders.ts` helper-file shape "when a command grows beyond a single handler file," and that the `.format.ts` / `.encoders.ts` files be **pure functions with no Effect or service dependencies** so they're unit-testable. `backups` already follows this (`backups.errors.ts`, `backups.format.ts` + `backups.format.unit.test.ts`, `backups.encoders.ts`). Here all of it lives in the handler, covered only through integration tests.

Concretely:
- Move the parser + encoders to `list.encoders.ts` and the timestamp helper to `list.format.ts` (pure, no Effect/services), and add `*.unit.test.ts` for the Go-parity edge cases — zero-values, null element, top-level null, omitempty omission, HTML / U+2028–U+2029 escaping, and JSON/YAML/TOML key ordering. These are textbook unit-test material and would shrink the integration file substantially.
- Move the error classes to `list.errors.ts`.

**Hoist consideration**

This is the first ported `list` command that needed Go's lenient-but-typed `json.Unmarshal` semantics, which the strict Effect schema can't reproduce, hence the bespoke parser. Every future list port (`projects list`, `secrets list`, etc.) will hit the same wall. Rather than re-inlining ~350 lines each time, it's worth considering a reusable "Go-parity decode + encode" helper in `legacy/shared/` e.g. an `omitempty` / HTML-escape option on the shared `encodeGoJson`, plus a generic Go-struct decoder. This is exactly the "Hoist Before You Duplicate" scenario the guide calls out. Fine to do as a follow-up, but flagging it now so the pattern doesn't get copy-pasted into the next port.

Comment (7ttp) [reply apps/cli/src/legacy/commands/functions/list/list.handler.ts:1]: done, refactored 💚

**Review by avallete — APPROVED** [review]
